### ISSUE TITLE
results.png not generated with python API

### ISSUE THREAD

**Issue by robmarkcole** [ultralytics/ultralytics#420]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

Using `yolo task=detect mode=train ..` the file `{HOME}/runs/detect/train2/results.png` is created, but using the python API it is not created (assume it should be).

### Environment

- Colab
- Ultralytics YOLOv8.0.6 🚀 Python-3.8.16 torch-1.13.0+cu116 CUDA:0 (Tesla T4, 15110MiB)

### Minimal Reproducible Example

```python
model = YOLO('yolov8m.pt')  # load a pretrained model (recommended for training)

# Use the model
results = model.train(
    data=f'{dataset.location}/data.yaml', 
    epochs=100,
    imgsz=640,
    batch=16
)  # train the model
```

### Additional

I am following https://github.com/roboflow/notebooks/blob/main/notebooks/train-yolov8-object-detection-on-custom-dataset.ipynb

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by pderrenger** [comment]
@robmarkcole hi there! 👋 It seems like the `results.png` file is not being generated when using the YOLOv8 Python API for training, unlike when running the command line interface (CLI) directly. This issue might stem from the differences in how the CLI and Python API handle the training process.

To address this, you can utilize the Python API to manually save the training results, such as loss curves and other metrics, during or after training. This can be achieved by leveraging the built-in logging and visualization functionalities provided by Ultralytics' YOLOv8 Python API.

If you encounter any difficulties implementing this approach or have further questions, feel free to reach out for more detailed guidance. Your feedback is essential to improving the user experience, and we appreciate your willingness to help. 🌟
